packages/arb-avm-cpp/data_storage/src/arbcore.cpp

```diff
@@ -150,8 +148,16 @@ rocksdb::Status ArbCore::initialize(const LoadedExecutable& executable) {
     // Use latest existing checkpoint
     ValueCache cache{1, 0};
 
-    auto status = reorgToMessageCountOrBefore(0, true, cache);
+    auto message_count_result = unexpiredMessageCount();
```

Comment:
With the refactor suggested to `reorgToMessageCountOrBefore` I think we could remove `unexpiredMessageCount` and instead reorg with a `output.last_inbox_timestamp < expired_timestamp` check